sdk/Sdk.Generators/FunctionMetadataProviderGenerator/FunctionMetadataProviderGenerator.Parser.cs

```diff
@@ -120,7 +121,8 @@ private bool TryGetBindings(MethodDeclarationSyntax method, SemanticModel model,
                     }
                     else if (!supportsRetryOptions && retryOptions is not null)
                     {
-                        // TODO: Log retry options related diagnostic error here
+                        _context.ReportDiagnostic(Diagnostic.Create(DiagnosticDescriptors.InvalidRetryOptions, method.GetLocation()));
```

Comment:
The expression `(retryOptions is not null` can be removed from 1118 and 122 since 116 does that.

```csharp
if (retryOptions is not null)
{
    if (supportsRetryOptions)
    {
        validatedRetryOptions = retryOptions;
    }
    else
    {
        _context.ReportDiagnostic(Diagnostic.Create(DiagnosticDescriptors.InvalidRetryOptions, method.GetLocation()));
        return false;
    }
}
```